src/main/java/gregtech/common/metatileentities/multi/multiblockpart/MetaTileEntityFluidHatch.java

```diff
@@ -180,23 +201,97 @@ protected ModularUI createUI(EntityPlayer entityPlayer) {
     }
 
     public ModularUI.Builder createTankUI(IFluidTank fluidTank, String title, EntityPlayer entityPlayer) {
+        // Create base builder/widget references
         Builder builder = ModularUI.defaultBuilder();
-        builder.image(7, 16, 81, 55, GuiTextures.DISPLAY);
-        TankWidget tankWidget = new TankWidget(fluidTank, 69, 52, 18, 18)
-                .setHideTooltip(true).setAlwaysShowFull(true);
-        builder.widget(tankWidget);
-        builder.label(11, 20, "gregtech.gui.fluid_amount", 0xFFFFFF);
-        builder.dynamicLabel(11, 30, tankWidget::getFormattedFluidAmount, 0xFFFFFF);
-        builder.dynamicLabel(11, 40, tankWidget::getFluidLocalizedName, 0xFFFFFF);
+        TankWidget tankWidget;
+
+        // Add input/output-specific widgets
+        if (isExportHatch) {
+            tankWidget = new PhantomTankWidget(fluidTank, 69, 43, 18, 18,
```

Comment:
I think this GUI would be better done similarly to the Quantum Tank gui, where the locked fluid is integrated into the fluid storage slot.